commit 4851d67896c2492cfcf761ae093aba6d49b88f29
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2019-06-13T06:49:43-06:00 (4 months ago)
    
    MANTA-4338 node-boray' s createBucket should unwrap the Fast data payload array

diff --git a/lib/buckets.js b/lib/buckets.js
index 0bc9642..a163143 100644
--- a/lib/buckets.js
+++ b/lib/buckets.js
@@ -55,7 +55,23 @@ function createBucket(rpcctx, owner, bucket, vnode, callback) {
         rpcargs: [arg],
         ignoreNullValues: true,
         log: log
-    }, callback);
+    }, function (err, buckets) {
+        if (err) {
+            callback(err);
+            return;
+        }
+
+        assert.array(buckets, 'buckets');
+
+        if (buckets.length !== 1) {
+            err = new VError('bad server response: expected 1 bucket, found %d',
+                buckets.length);
+            callback(err);
+            return;
+        }
+
+        callback(null, buckets[0]);
+    });
 }
 
 function createBucketNoVnode(rpcctx, owner, bucket, callback) {
